src/utilities/filetypes/EpwFile.cpp

```diff
@@ -3967,6 +3974,31 @@ bool EpwFile::parse(std::istream& ifs, bool storeData) {
     }
   }
 
+  for (unsigned int i = 0; i < epw_strings.size(); i++) {
+    int year = epw_strings[i].year;
+    int month = epw_strings[i].month;
+    int day = epw_strings[i].day;
+    int hour = epw_strings[i].hour;
+    int currentMinute = epw_strings[i].currentMinute;
+
+    // Fix for #5214: for TMY files where 28-day Feb is from a leap year, we need xxxx-Mar-01 00:00:00 and not xxxx-Feb-29 00:00:00
+    // If the TMY file actually contained Feb 29, we still change xxxx-Feb-29 00:00:00 to xxxx-Mar-01 00:00:00, but it doesn't matter since you'd still get Bad Date on xxxx-Feb-29 01:00:00, etc.
+    if ((!realYear) && (month == 2) && (day == 28) && (hour == 24) && (currentMinute == 0)) {
+      Date date(month, day, year);
+      if (date.isLeapYear()) {
+        day = 29;
+      }
+    }
+
+    boost::optional<EpwDataPoint> pt = EpwDataPoint::fromEpwStrings(year, month, day, hour, currentMinute, epw_strings[i].strings);
+    if (pt) {
+      m_data.push_back(pt.get());
```

Comment:
Here the temporary `pt` makes sense to me since you have the `push_back` protected by a conditional.

Reply:
No action.

Reply:
`m_data.emplace_back(std::move(*pt))`